src/library/coverartcache.cpp

```diff
@@ -7,9 +7,42 @@
 #include "library/coverartcache.h"
 #include "library/coverartutils.h"
 
-// Large cover art wastes space in our cache when we typicaly won't show them at
-// their full size. If no width is specified, this is the maximum width cap.
-const int kMaxCoverWidth = 300;
+
+namespace {
+    QString pixmapCacheKey(quint16 hash, int width) {
+        if (width == 0) {
+            return QString("CoverArtCache_%1").arg(QString::number(hash));
+        } else {
+            return QString("CoverArtCache_%1_%2")
+                    .arg(QString::number(hash)).arg(width);
+        }
+    }
+
+    // Large cover art wastes space in our cache when we typically won't show them at
+    // their full size. If no width is specified, this is the maximum width cap.
+    const int kMaxCoverSize = 300;
+
+    // The transformation mode when scaling images
+    const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
+
+    // Resizes the image (preserving aspect ratio) if it is larger than
+    // maxEdgeSize on either side.
+    inline QImage limitImageSize(const QImage& image, int maxSize) {
```

Comment:
Sorry for the extra work, Daniel. I factored out the first move as a 
separate commit, but missed to also split the second one into two parts.

At least the big functions have not changed at all. I remember that I 
changed an "if { return } return" into the equivalent "if { return } else { 
return }", but the rest of the code should be identical.

On 07/31/2015 11:37 AM, Daniel Schürmann wrote:
>
> In src/library/coverartcache.cpp 
> <https://github.com/mixxxdj/mixxx/pull/662#discussion_r35959513>:
>
> > +        } else {
> > +            return QString("CoverArtCache_%1_%2")
> > +                    .arg(QString::number(hash)).arg(width);
> > +        }
> > +    }
> > +
> > +    // Large cover art wastes space in our cache when we typically won't show them at
> > +    // their full size. If no width is specified, this is the maximum width cap.
> > +    const int kMaxCoverSize = 300;
> > +
> > +    // The transformation mode when scaling images
> > +    const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
> > +
> > +    // Resizes the image (preserving aspect ratio) if it is larger than
> > +    // maxEdgeSize on either side.
> > +    inline QImage limitImageSize(const QImage& image, int maxSize) {
>
> Since it is "only" a 280 line PR this will be not be required.
> I will read it line by line. For similar PRs it would be nice to have the 
> moves separate.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/662/files#r35959513>.
>